bib/cmd/bootc-image-builder/main.go

```diff
@@ -529,25 +529,55 @@ func rootPreRunE(cmd *cobra.Command, _ []string) error {
 	return nil
 }
 
-// TODO: provide more version info (like actual version number) once we
-// release a real version
-func cmdVersion(_ *cobra.Command, _ []string) error {
+func cmdVersion() (string, error) {
 	info, ok := debug.ReadBuildInfo()
 	if !ok {
-		return fmt.Errorf("cannot read build info")
+		return "", fmt.Errorf("cannot read build info")
 	}
 	var gitRev string
+	var buildTime string
+	var buildTainted bool
+	ret := []string{}
 	for _, bs := range info.Settings {
 		if bs.Key == "vcs.revision" {
 			gitRev = bs.Value
-			break
+			continue
+		}
+		if bs.Key == "vcs.time" {
+			buildTime = bs.Value
+			continue
+		}
+		if bs.Key == "vcs.modified" {
+			bT, err := strconv.ParseBool(bs.Value)
+			if err != nil {
+				logrus.Errorf("Error parsing 'vcs.modified': %v", err)
+				bT = true
+			}
+
+			buildTainted = bT
+			continue
 		}
 	}
 	if gitRev != "" {
-		fmt.Printf("revision: %s\n", gitRev[:7])
+		ret = append(ret, fmt.Sprintf("revision: %s", gitRev[:7]))
 	} else {
-		fmt.Printf("revision: unknown\n")
+		ret = append(ret, "revision: unknown")
+	}
+	if buildTime != "" {
```

Comment:
You are right, this should be more a compromise between proper version number and gitSHA and it was mentioned in some of our discussions, that this would be helpful to just also show the time.
As it's clearly marked as "build time" there should be not too many confusions.

Reply:
please check